docs/CLIUsage.md

```diff
@@ -1,23 +1,120 @@
 # CLI Usage
 
-This document demonstrates how to use the CLI but also shows what happens in `Kudo` under the hood, which can be helpful when troubleshooting.
+This document demonstrates how to use the CLI but also shows what happens in `KUDO` under the hood, which can be helpful when troubleshooting.
+
+## Table of Contents
+
+   * [CLI Usage](#cli-usage)
+      * [Table of Contents](#table-of-contents)
+      * [Kubectl KUDO Plugin](#kubectl-kudo-plugin)
+         * [Requirements](#requirements)
+         * [Install](#install)
+         * [Commands](#commands)
+         * [Flags](#flags)
+         * [Environment Variables](#environment-variables)
+         * [Examples](#examples)
+            * [Install a Framework](#install-a-framework)
+               * [Install just the Framework without dependencies](#install-just-the-framework-without-dependencies)
+            * [Install a Framework with Dependencies](#install-a-framework-with-dependencies)
+            * [Install a Framework with InstanceName &amp; Parameters](#install-a-framework-with-instancename--parameters)
+            * [List Instances](#list-instances)
+            * [Get the Status of an Instance](#get-the-status-of-an-instance)
+            * [Get the History to PlanExecutions](#get-the-history-to-planexecutions)
+
+## Kubectl KUDO Plugin
+
+### Requirements
+
+- KUDO client automatically created via `code-generator`
+- GitHub set up
+  - GitHub OTP token in `$HOME/.git-credentials` OR
+  - GitHub Basic Auth via `GIT_USER` and `GIT_PASSWORD` environment variables exposed
 
-There are two CLIs you can use at the moment:
+### Install
+
+Install the plugin from your `$GOPATH/src/github.com/kudobuilder/kudo` root folder via:
+
+- `go install ./cmd/kubectl-kudo`
+
+### Commands
+
+|  Syntax | Description  | 
+|---|---|
+| `kubectl kudo install <name> [flags]`  |  Installs a Framework from the official [KUDO repo](https://github.com/kudobuilder/frameworks). |
+| `kubectl kudo list instances [flags]` | Show all available instances. | 
+| `kubectl kudo plan status [flags]` | View all available plans. |
+| `kubectl kudo plan history <name> [flags]` | View all available plans. |
+
+### Flags
```

Comment:
I am not so much concerned about `flags` than I am for `examples`. Cobra generates flags usage [automatically](https://github.com/spf13/cobra#usage-message) for us. There could be maybe a problem with GlobalFlags that will be overriden (but e.g. kubelet is using a [fix](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L270-L284) for it that we could then implement - I am using it partly already [here](https://github.com/kudobuilder/kudo/pull/138#discussion_r263927634) to fix the error output in combination with usage). So when the flag output is automatically generated by cobra during usage via `-h/--help` we should just look at `examples`.

Right now they are defined as a variable e.g. in `cmd/install.go`. We could e.g. use `go generate` to generate a `install_example.go` file within the `cmd` package to cover the example display while using the CLI. 

To cover that also the docs will be automatically updated we could generate the `CLIUsage.md` in general. That could be maybe an entire new thing in general of generating and updating documentation automatically that also would tie in how it will then be automatically displayed on kudo.dev ...

Reply:
I agree it would be annoying at this point.  Where do you want to keep track of the request to automate this generation?  An issue, or a proposal in the CICD Kep?